source/common/upstream/upstream_impl.h

```diff
@@ -387,6 +388,9 @@ class HostSetImpl : public HostSet {
       HostsPerLocalityConstSharedPtr all_hosts_per_locality,
       LocalityWeightsConstSharedPtr locality_weights, uint32_t overprovisioning_factor);
 
+  static absl::optional<uint32_t>
+  chooseLocality(std::unique_ptr<EdfScheduler<LocalityEntry>>& locality_scheduler);
+
   std::vector<std::shared_ptr<LocalityEntry>> locality_entries_;
   std::unique_ptr<EdfScheduler<LocalityEntry>> locality_scheduler_;
```

Comment:
Should we also rename these to `healthy_locality_entries_` and `healthy_locality_entries_` respectively?